### DISCUSSION TITLE
Use $event.target.value in @input in template

### DISCUSSION THREAD

**Discussion by atlansien** [nuxt/nuxt#20904] [Questions]
Example
```
<script setup lang="ts">
  const updateValue = (evt) => {
    console.log(evt);
  }
</script>

<template>
  <div>
    <input @input="updateValue($event.target.value)" />
  </div>
</template>
```
In this case, the following error occurs.
```
// => Property 'value' does not exist on type 'EventTarget'. ts(2339)
```
To solve the problem, I now write the following.
```
<input @input="updateValue(($event.target as HTMLInputElement).value)" />
```

Do you have a better idea?

**Comment by ndragun92** [comment]
```
<script setup lang="ts">
  const updateValue = (event) => {
    console.log(event?.target?.value);
  }
</script>

<template>
  <div>
    <input @input="updateValue" />
  </div>
</template>
```

This is how I usually do it

https://stackblitz.com/edit/github-pgdo4d?file=app.vue

**Reply by atlansien** [reply]
@ndragun92 

Thank you very much!
I just want to use it within the template myself!

**Reply by ndragun92** [reply]
It is not recommended to use it within thw template. Way harder to make unit tests and maintain code. If u have to emit value then u can emit it via template
